math_parser.py

```diff
@@ -69,13 +76,15 @@ def paren(string):
 
 
 def expression(tree):
+    # Handles scalar constants and resolves names at runtime
     if not isinstance(tree, list):
         try:
             f = float(tree)
             return lambda s: f
         except:
             n = tree
             return lambda s: s[n]
+
     if len(tree) == 1:
         return expression(tree[0])
     for op in EMDAS.keys():
```

Comment:
Are we relying on EMDA.keys() giving the ops in the correct order here? Seems fishy~